pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -106,7 +106,7 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 			}
 
 			for _, status := range pod.Status.ContainerStatuses {
-				if status.Name == rev.Spec.GetContainer().Name {
+				if status.Name != resources.QueueContainerName {
```

Comment:
You'll need to adjust the `break` on line `123`. Otherwise if the failing sidecar container is second in the containerStatuses list it won't fail.

Reply:
I think we don't need to use `break` anymore, but I changed the position of `break` for fast fail.